vars/buildPlugin.groovy

```diff
@@ -102,6 +106,35 @@ def call(Map params = [:]) {
                     '-Dcheckstyle.failOnViolation=false',
                     '-Dcheckstyle.failsOnError=false',
                   ]
+                  if (artifactCachingProxyEnabled) {
+                    withCredentials([
+                      usernamePassword(credentialsId: 'artifact-caching-proxy-credentials', usernameVariable: 'ARTIFACT_CACHING_PROXY_USERNAME', passwordVariable: 'ARTIFACT_CACHING_PROXY_PASSWORD')
+                    ]) {
+                      final String defaultProxyProvider = 'azure'
+                      def availableProxyProviders = ['azure', 'do', 'aws']
+                      String requestedProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER
+                      String mavenSettings = libraryResource 'artifact-caching-proxy-settings.xml'
+
+                      // As azure VM agents don't have this env var, setting a default provider if none is specified or if the provider isn't available
+                      if (requestedProvider == null || requestedProvider == '' || !availableProxyProviders.contains(requestedProvider)) {
+                        requestedProvider = defaultProxyProvider
+                        echo "INFO: no valid artifact caching proxy provider specified, set to '$defaultProxyProvider' by default."
+                      }
+
+                      echo "Setting up Maven to use artifact caching proxy from '${requestedProvider}' provider..."
+
+                      settingsFile = "${pwd tmp: true}/mavenSettings.xml"
+                      mavenSettings = mavenSettings.replace('PROVIDER', requestedProvider)
+                      mavenSettings = mavenSettings.replace('SERVER-USERNAME', env.ARTIFACT_CACHING_PROXY_USERNAME)
+                      mavenSettings = mavenSettings.replace('SERVER-PASSWORD', env.ARTIFACT_CACHING_PROXY_PASSWORD)
+                      writeFile file: settingsFile, text: mavenSettings
```

Comment:
Still not ideal since you are loading the password into the Groovy program state. Better to do replacements with `sed`.

Reply:
> Still not ideal since you are loading the password into the Groovy program state.

I didn't knew it could be problematic.

> Better to do replacements with sed.

Will look into it, thanks for the feedback and alternative script suggestion!